### DISCUSSION TITLE
How do i get genQL library to work with Nuxt3

### DISCUSSION THREAD

**Discussion by shiftlabs1** [nuxt/nuxt#17354] [Questions]
I tried using genql (github.com/remorses/genql) with nuxt3 (I have it working Ok on a vue3 project already), but get the error below 

```
 ERROR  Rollup error: Unexpected token (Note that you need plugins to import files that are not JavaScript)   nitro 00:06:13


 ERROR  Unexpected token (Note that you need plugins to import files that are not JavaScript)                       00:06:13

  at error (node_modules/rollup/dist/es/shared/rollup.js:1858:30)
  at Module.error (node_modules/rollup/dist/es/shared/rollup.js:12424:16)
  at Module.tryParse (node_modules/rollup/dist/es/shared/rollup.js:12801:25)
  at Module.setSource (node_modules/rollup/dist/es/shared/rollup.js:12706:24)
  at ModuleLoader.addModuleSource (node_modules/rollup/dist/es/shared/rollup.js:22180:20)
```
When searching for similar issues online, I find that the stack traces always shows the line in your code where the error is thrown, but this is not the case in Nuxt3. This and other errors I found only shows the stack trace from rollup or nitro with no information about what part of your application is responsible. I had to keep eliminating modules from a relatively large code base to get to this.
Nitro Server (used in Nuxt3) uses rollup under the hood but i don't know what configuration is needed in nuxtconfig or anywhere else to make this error go away. when i take out all references to the genQL client this error goes away

**Comment by Flanker72** [comment]
Hope [my article about GraphQL](https://medium.com/@flanker72/nuxt3-complex-solutions-graphql-integration-664e24e48cef) may help you to find the solution.
